### ISSUE TITLE
crontab not running periodic task

### ISSUE THREAD

**Issue by mymtw** [celery/celery#4767]
## Checklist

- [x ] celery==4.1.1. and Ubuntu 16.04 and Python 3.5.2
- [-] I have verified that the issue exists against the `master` branch of Celery.

## Steps to reproduce
```
from celery import Celery
from celery.schedules import crontab

# This settings will be overrided
RABBITMQ_HOST = ''
RABBITMQ_PORT = ''
RABBITMQ_USERNAME = ''
RABBITMQ_PASSWORD = ''


class Config:
    timezone = 'Europe/Minsk'
    enable_utc = True

    task_always_eager = False
    broker_url = 'amqp://%s:%s@%s:%s//' % (RABBITMQ_USERNAME, RABBITMQ_PASSWORD, RABBITMQ_HOST, RABBITMQ_PORT)
    result_expires = 3600
    worker_prefetch_multiplier = 1
    worker_max_memory_per_child = 250000  # 250MB

    imports = (
        'app.jobs.clients_job',
    )
    task_routes = {
        'app.jobs.clients_job.send_email': {'queue': 'clients'},
        # Periodic tasks
        'app.jobs.clients_job.send_email_notifications': {'queue': 'periodic_api'},
    }

    beat_schedule = {
        'send-email-notifications': {
            'task': 'app.jobs.clients_job.send_email_notifications',
            'schedule': crontab(hour=16, minute=4),
            'args': (),
        },
    }
    worker_hijack_root_logger = False
    accept_content = ['json']


celery = Celery()
celery.config_from_object(Config)

############## `app.jobs.clients_job`
@celery.task
def send_email_notifications():
    t = datetime.date.today()
    logger.info('started')
    logger.info(t)

Run
$ celery worker -c 1 -n prtperiodicapi -l INFO -A config.celery_config -B -E -Q periodic_api -s /tmp/api_celerybeat_schedule_periodic

```
## Expected behavior
Expected running of task `send_email_notifications` runnig at 16:04 hours
## Actual behavior
task `send_email_notifications` not runnig at 16:04 hours

**Comment by mymtw** [comment]
@auvipy maybe you can help with this issue?

**Comment by juannyG** [comment]
You need to run an instance of `celery beat` along side the worker. `beat` will evaluate the `beat_schedule` portion of the configuration, see: http://docs.celeryproject.org/en/latest/userguide/configuration.html#beat-schedule

I changed paths a little bit to keep things simple:
### celery_config.py
```
import datetime

from celery import Celery
from celery.schedules import crontab

class Config:
    timezone = 'Europe/Minsk'
    enable_utc = True

    task_always_eager = False
    broker_url = 'redis://:@127.0.0.1:6379/13'
    result_expires = 3600
    worker_prefetch_multiplier = 1
    worker_max_memory_per_child = 250000  # 250MB

    beat_schedule = {
        'send-email-notifications': {
            'task': 'celery_config.send_email_notifications',
            'schedule': crontab(),
            'args': (),
        },
    }
    worker_hijack_root_logger = False
    accept_content = ['json']


app = Celery()
app.config_from_object(Config)

@app.task
def send_email_notifications():
    t = datetime.date.today()
    print('started')
    print(t)
```

### worker execution
```
(celery-issues-env) j@zebra:~/celery-issues-env/celery-issues$ celery worker -a celery_config --loglevel=info
 
 -------------- celery@zebra v4.1.1 (latentcall)
---- **** ----- 
--- * ***  * -- linux-4.4.0-124-generic-x86_64-with-debian-stretch-sid 2018-05-26 16:24:51
-- * - **** --- 
- ** ---------- [config]
- ** ---------- .> app:         __main__:0x7f15f8f9c978
- ** ---------- .> transport:   redis://127.0.0.1:6379/13
- ** ---------- .> results:     disabled://
- *** --- * --- .> concurrency: 4 (prefork)
-- ******* ---- .> task events: off (enable -e to monitor tasks in this worker)
--- ***** ----- 
 -------------- [queues]
                .> celery           exchange=celery(direct) key=celery
                

[tasks]
  . celery_config.send_email_notifications

[2018-05-26 16:24:51,895: info/mainprocess] connected to redis://127.0.0.1:6379/13
[2018-05-26 16:24:51,902: info/mainprocess] mingle: searching for neighbors
[2018-05-26 16:24:52,916: info/mainprocess] mingle: all alone
[2018-05-26 16:24:52,924: info/mainprocess] celery@zebra ready.
```

At this point - the worker was sitting idly. Then, I opened another terminal and ran a `beat` instance:

### beat execution
```
(celery-issues-env) j@zebra:~/celery-issues-env/celery-issues$ celery beat -A celery_config --loglevel=INFO
celery beat v4.1.1 (latentcall) is starting.
__    -    ... __   -        _
LocalTime -> 2018-05-26 16:23:39
Configuration ->
    . broker -> redis://127.0.0.1:6379/13
    . loader -> celery.loaders.app.AppLoader
    . scheduler -> celery.beat.PersistentScheduler
    . db -> celerybeat-schedule
    . logfile -> [stderr]@%INFO
    . maxinterval -> 5.00 minutes (300s)
[2018-05-26 16:23:39,926: INFO/MainProcess] beat: Starting...
```
I defined my schedule with `crontab()` so it would run every minute, and as expected, here's the output from both my `beat` and `worker` instances

### beat output
```
[2018-05-26 16:24:00,067: INFO/MainProcess] Scheduler: Sending due task send-email-notifications (celery_config.send_email_notifications)
[2018-05-26 16:25:00,039: INFO/MainProcess] Scheduler: Sending due task send-email-notifications (celery_config.send_email_notifications)
[2018-05-26 16:26:00,007: INFO/MainProcess] Scheduler: Sending due task send-email-notifications (celery_config.send_email_notifications)
[2018-05-26 16:27:00,048: INFO/MainProcess] Scheduler: Sending due task send-email-notifications (celery_config.send_email_notifications)
[2018-05-26 16:28:00,093: INFO/MainProcess] Scheduler: Sending due task send-email-notifications (celery_config.send_email_notifications)
```
### worker output
```
[2018-05-26 16:25:00,041: INFO/MainProcess] Received task: celery_config.send_email_notifications[c257027c-13f1-4d13-ad88-95578c6164b9]  
[2018-05-26 16:25:00,043: WARNING/ForkPoolWorker-4] started
[2018-05-26 16:25:00,044: WARNING/ForkPoolWorker-4] 2018-05-26
[2018-05-26 16:25:00,044: INFO/ForkPoolWorker-4] Task celery_config.send_email_notifications[c257027c-13f1-4d13-ad88-95578c6164b9] succeeded in 0.0012485550250858068s: None
[2018-05-26 16:26:00,008: INFO/MainProcess] Received task: celery_config.send_email_notifications[e0aea10a-1d99-4826-b061-39214040a14c]  
[2018-05-26 16:26:00,009: WARNING/ForkPoolWorker-3] started
[2018-05-26 16:26:00,009: WARNING/ForkPoolWorker-3] 2018-05-26
[2018-05-26 16:26:00,010: INFO/ForkPoolWorker-3] Task celery_config.send_email_notifications[e0aea10a-1d99-4826-b061-39214040a14c] succeeded in 0.0006158810574561357s: None
[2018-05-26 16:27:00,050: INFO/MainProcess] Received task: celery_config.send_email_notifications[ce36c8ec-d59e-41d4-ac0d-947fc8ce84e2]  
[2018-05-26 16:27:00,050: WARNING/ForkPoolWorker-4] started
[2018-05-26 16:27:00,050: WARNING/ForkPoolWorker-4] 2018-05-26
[2018-05-26 16:27:00,050: INFO/ForkPoolWorker-4] Task celery_config.send_email_notifications[ce36c8ec-d59e-41d4-ac0d-947fc8ce84e2] succeeded in 0.0003398279659450054s: None
[2018-05-26 16:28:00,095: INFO/MainProcess] Received task: celery_config.send_email_notifications[432177e4-4dbf-42a5-9541-41c058b185f0]  
[2018-05-26 16:28:00,096: WARNING/ForkPoolWorker-3] started
[2018-05-26 16:28:00,097: WARNING/ForkPoolWorker-3] 2018-05-26
[2018-05-26 16:28:00,097: INFO/ForkPoolWorker-3] Task celery_config.send_email_notifications[432177e4-4dbf-42a5-9541-41c058b185f0] succeeded in 0.0005777250044047832s: None
```

I think this issue can be closed.

**Comment by mymtw** [comment]
@juannyG thank you, but this setting`crontab()` ( running each minute ) worked and earlier, it not working if I'm providing specific time `crontab(hour=0, minute=31)`

**Comment by juannyG** [comment]
Sorry for not testing that! After setting some values in the `crontab` method the schedule worked as expected only if I scheduled against **UTC**

This seems related to issue https://github.com/celery/celery/issues/943

**Comment by juannyG** [comment]
No - `enabled_utc` was removed, and thus defaulted to `True` - that was the only way I could get this to work.

It also almost seems as though: if `enabled_utc` was `True` - timezone is simply ignored.

I don't know if it'll be fixed...but I'm going to continue looking into this at least.

**Comment by juannyG** [comment]
I just tested against the master branch of `celery` and it works. It uses the timezone I define, regardless of `enable_utc` value and crontab runs as expected.

Looks like it may be slotted for a `v4.2.0` release? That's the release candidate reference in the master branch, for now: https://github.com/celery/celery/blob/master/celery/__init__.py#L17

**Comment by mymtw** [comment]
anyway it's dangerously to install `RC` version in project, I will test it when `4.2.0` will be released then will close this issue

**Comment by mymtw** [comment]
THIS ~~IS SPARTA~~ has been fixed in `v4.2.0`
thx
